File: src/robot.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

// saved for reference only

#include "robot.hpp"
#include "lifter.hpp" // Include the Lifter header file

#include <frc2/command/CommandScheduler.h>

Robot::Robot() {}

void Robot::RobotPeriodic() {
    frc2::CommandScheduler::GetInstance().Run();
}

void Robot::DisabledInit() {}

void Robot::DisabledPeriodic() {}

void Robot::DisabledExit() {}

void Robot::AutonomousInit() {
    m_autonomousCommand = m_container->GetAutonomousCommand();

    if (m_autonomousCommand) {
        m_autonomousCommand->Schedule();
    }
}

void Robot::AutonomousPeriodic() {}

void Robot::AutonomousExit() {}

void Robot::TeleopInit() {
    if (m_container == nullptr) {
        m_container.reset (new RobotContainer());
    }
    if (m_autonomousCommand) {
        m_autonomousCommand->Cancel();
    }
}

void Robot::TeleopPeriodic() {
    m_container->drivetrain.Periodic();
    m_container->lifter.Periodic(); // Call the Lifter's Periodic method
}

void Robot::TeleopExit() {
}

void Robot::TestInit() {
    frc2::CommandScheduler::GetInstance().CancelAll();
}

void Robot::TestPeriodic() {}

void Robot::TestExit() {}

#ifndef RUNNING_FRC_TESTS
int main() {
    return frc::StartRobot<Robot>();
}
#endif

File: pybot/robot.py
#!/usr/bin/env python3
#
# Copyright (c) FIRST and other WPILib contributors.
# Open Source Software; you can modify and/or share it under the terms of
# the WPILib BSD license file in the root directory of this project.
#

import typing
import wpilib, commands2

import autos
from robotcontainer import RobotContainer

LATENCY_SECONDS = 0.03

class MyRobot(wpilib.TimedRobot):
    autonomousCommand: typing.Optional[commands2.Command] = None
    chooser: None
    
    def __init__(self):
        super().__init__(LATENCY_SECONDS)
    
    def robotInit(self) -> None:
        self.container = RobotContainer()
        self.scheduler = commands2.CommandScheduler.getInstance()
        self.registerTrajectories()

    def registerTrajectories(self) -> None:
        self.chooser = autos.createChooser()
        wpilib.SmartDashboard.putData ('Trajectory Files', self.chooser)

    def selectedTrajectory(self) -> str:
        return self.chooser.getSelected()

    def robotPeriodic(self) -> None:
        self.scheduler.run()

    def disabledInit(self) -> None:
        pass

    def disabledPeriodic(self) -> None:
        pass

    def autonomousInit(self) -> None:
        selected = self.selectedTrajectory()
        self.autonomousCommand = self.container.getAutonomousCommand (selected)
        if self.autonomousCommand:
            self.autonomousCommand.schedule()

    def autonomousPeriodic(self) -> None:
        pass

    def teleopInit(self) -> None:
        if self.autonomousCommand:
           self.autonomousCommand.cancel()
           self.autonomousCommand = None

        self.container.configureButtonBindings()

    def teleopPeriodic(self) -> None:
        pass

    def testInit(self) -> None:
        self.scheduler.cancelAll()

    def simulationInit(self) -> None:
        pass

    def simulationPeriodic(self) -> None:
        pass
